framework-docs/src/docs/asciidoc/web/webflux-webclient.adoc

```diff
@@ -1027,19 +1027,16 @@ any response content, whether expected or not, is released:
 [source,kotlin,indent=0,subs="verbatim,quotes",role="secondary"]
 .Kotlin
 ----
-	fun renewTokenFilter(): ExchangeFilterFunction? {
-		return ExchangeFilterFunction { request: ClientRequest?, next: ExchangeFunction ->
-			next.exchange(request!!).flatMap { response: ClientResponse ->
+	fun renewTokenFilter(): ExchangeFilterFunction {
+		return ExchangeFilterFunction { request: ClientRequest, next: ExchangeFunction ->
+			mono {
```

Comment:
I think `mono(Dispatchers.Unconfined)` should be used here instead of `mono` which may default to `Dispatchers.Default` which is baked by a shared thread pool. That would also be consistent by `CoWebFilter` introduced by @poutsma.

Reply:
Also, why can't I find the suspend function in CoWebFilter documented anywhere? https://docs.spring.io/spring-framework/docs/current/kdoc-api/spring-web/org.springframework.web.server/-co-web-filter/index.html

Reply:
> Are there any plans for a CoExchangeFilterFunction?
See me answer in the PR level comment.

> Also, why can't I find the suspend function in CoWebFilter documented anywhere?
`CoWebFilter` is documented in the API documentation you linked, and is "just" a `WebFilter` implementation. I don't think the reference documentation includes a `WebFilter` implementation code sample where we should translate the Kotlin version with `CoWebFilter`.

Reply:
I'd expect that a protected method intended to be overridden like https://github.com/spring-projects/spring-framework/blob/f7367ceae53f6f812b8634fc324933aaf63785c9/spring-web/src/main/kotlin/org/springframework/web/server/CoWebFilter.kt#L48 would show up in the kdoc, but I might just be wrong about that

Reply:
Good catch, maybe a Dokka issue for that specific use case where we have 2 `filter` functions pretty close in term of signature.

Can you please create a related issue in https://github.com/Kotlin/dokka linking this source code and the generated kdoc?